<launch>

  <arg name="simulation" default="$(eval env('RUN_TYPE') == 'simulation')" />

  <arg name="uav_name" default="$(optenv UAV_NAME uav)"/>
  <arg name="respawn" default="false" />

  <arg name="rosbag_remap" default="false" />
  <arg unless="$(arg rosbag_remap)" name="suf" value=""/>
  <arg     if="$(arg rosbag_remap)" name="suf" value="_"/>

  <arg name="debug" default="false" />
  <arg unless="$(arg debug)" name="launch_prefix" value=""/>
  <arg     if="$(arg debug)" name="launch_prefix" value="debug_roslaunch"/>

  <arg name="profile" default="false" />
  <arg unless="$(arg profile)" name="launch_prefix_profile" value=""/>
  <arg     if="$(arg profile)" name="launch_prefix_profile" value="valgrind --tool=callgrind --callgrind-out-file='callgrind.out.%p'"/>

  <!-- nodelet-related arguments -->
  <arg name="standalone" default="true" />
  <arg name="manager" default="swarm_nodelet_manager" />
  <arg name="swarm_nodelet" value="difec_ron/SwarmControl" />
  <arg unless="$(arg standalone)" name="nodelet_load" value="load $(arg swarm_nodelet) $(arg manager)"/>
  <arg     if="$(arg standalone)" name="nodelet_load" value="standalone $(arg swarm_nodelet)"/>

  <group ns="$(arg uav_name)">

    <node pkg="nodelet" type="nodelet" name="swarm_control" args="$(arg nodelet_load)" output="screen" launch-prefix="$(arg launch_prefix) $(arg launch_prefix_profile)" respawn="$(arg respawn)" >

      <rosparam file="$(find difec_ron)/config/swarm_control.yaml" />

      <param name="uav_name" value="$(arg uav_name)" />
      <param name="world_frame_id" value="$(arg uav_name)/rtk_origin" />
    
      <!-- Subscribers -->
      <remap from="~detections_in" to="uvdar/filteredPoses" />

      <!-- Publishers -->

    </node>
  
  </group>

</launch>
